curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/650756e20cffbe41305a0dde.md

```diff
@@ -0,0 +1,435 @@
+---
+id: 650756e20cffbe41305a0dde
+title: Step 110
+challengeType: 0
+dashedName: step-110
+---
+
+# --description--
+
+Inside the condition, you want to first set the `isCheckpointCollisionDetectionActive` to false.
+
+Then you will need to call the `showCheckpointScreen` function and pass in the string `"You reached the final checkpoint!"` as an argument.
+
+Lastly, you will need to call the `movePlayer` function and pass in the string `"ArrowRight"` as the first argument, the number `0` as the second argument, and the boolean `false` as the third argument.
+
+# --hints--
+
+Test 1
+
+```js
+
+```
```

Comment:
These work:

You should set `isCheckpointCollisionDetectionActive` to `false` inside the `if` statement.

```js
assert.match(code, /if\s*\(\s*index\s*===\s*checkpoints\.length\s*-\s*1\s*\)\s*\{\s*isCheckpointCollisionDetectionActive\s*=\s*false;?\s*/)
```

You should call the `showCheckpointScreen` function and pass in `"You reached the final checkpoint!"` as an argument.

```js
assert.match(code, /if\s*\(\s*index\s*===\s*checkpoints\.length\s*-\s*1\s*\)\s*\{\s*isCheckpointCollisionDetectionActive\s*=\s*false;?\s*showCheckpointScreen\(("|'|`)You\s+reached\s+the\s+final\s+checkpoint!\1\);?/)
```